capsules/src/virtual_i2c.rs

```diff
@@ -244,22 +265,42 @@ impl i2c::I2CDevice for I2CDevice<'_> {
         }
     }
 
-    fn write_read(&self, data: &'static mut [u8], write_len: u8, read_len: u8) {
-        self.buffer.replace(data);
-        self.operation.set(Op::WriteRead(write_len, read_len));
-        self.mux.do_next_op();
+    fn write_read(
+        &self,
+        data: &'static mut [u8],
+        write_len: u8,
+        read_len: u8,
+    ) -> Result<(), (Error, &'static mut [u8])> {
+        if self.operation.get() == Op::Idle {
+            self.buffer.replace(data);
+            self.operation.set(Op::WriteRead(write_len, read_len));
+            self.mux.do_next_op();
+            Ok(())
+        } else {
+            Err((Error::ArbitrationLost, data))
+        }
```

Comment:
Just flagging this because it is clearly a change in logic -- the idea here is that if a node is not currently idle we will always reject new requests does make sense to me.

Reply:
I think this was an error before, the mux would accept new requests wile having one in progress. I think this would have lead to buffer leakage and unexpected results.

Reply:
That makes sense to me